packages/upload-api/src/blob/accept.js

```diff
@@ -60,3 +62,76 @@ export function blobAcceptProvider(context) {
     },
   })
 }
+
+/**
+ * Polls `blob/accept` whenever we receive a receipt. It may error if passed
+ * receipt is for `http/put` task that refers to the `blob/allocate` that we
+ * are unable to find.
+ *
+ * @param {API.ConcludeServiceContext} context
+ * @param {API.Receipt} receipt
+ * @returns {Promise<API.Result<{}, API.StorageGetError>>}
+ */
+export const poll = async (context, receipt) => {
+  const ran = Invocation.isInvocation(receipt.ran)
+    ? { ok: receipt.ran }
+    : await context.agentStore.invocations.get(receipt.ran)
+
+  // If can not find an invocation for this receipt there is nothing to do here,
+  // if it was receipt for `http/put` we would have invocation record.
+  if (!ran.ok) {
+    return { ok: {} }
+  }
+
+  // Detect if this is a receipt is for an `http/put` invocation
+  const put = /** @type {?API.HTTPPut} */ (
+    ran.ok.capabilities.find(({ can }) => can === HTTP.put.can)
+  )
+
+  // If it's not an http/put invocation nothing to do here.
+  if (put == null) {
+    return { ok: {} }
+  }
+
+  // Otherwise we are going to lookup allocation corresponding to this http/put
+  // in order to issue blob/accept.
+  const [, allocation] = /** @type {API.UCANAwait} */ (put.nb.url)['ucan/await']
+  const result = await context.agentStore.invocations.get(allocation)
+  // If count not lookup blob/allocate invocation there is something wrong in
+  // the system and we return error so it could be propagated to the user. It is
+  // not a proper solution as user can not really do anything, but still seems
+  // better than silently ignoring, this way user has a chance to report a
+  // problem. Clint test could potentially also catch errors.
+  if (result.error) {
+    return result
+  }
+
+  const [allocate] = /** @type {[API.BlobAllocate]} */ (result.ok.capabilities)
+
+  // If this is a receipt for the http/put we will perform blob/accept.
+  const blobAccept = await W3sBlob.accept.invoke({
+    issuer: context.id,
+    audience: context.id,
+    with: context.id.did(),
+    nb: {
+      blob: put.nb.body,
+      space: allocate.nb.space,
+      _put: {
+        'ucan/await': ['.out.ok', receipt.ran.link()],
+      },
+    },
+    // ⚠️ We need invocation to be deterministic which is why we use exact
+    // same as it is on allocation which will guarantee that expiry is the
+    // same regardless when we received `http/put` receipt.
+    //
+    // ℹ️ This works around the fact that we index receipts by invocation link
+    // as opposed to task link which would not care about the expiration.
+    expiration: result.ok.expiration,
+  })
+
+  // We do not care about the result we just want receipt to be issued and
+  // stored.
+  await blobAccept.execute(context.getServiceConnection())
+
```

Comment:
we do not want the task scheduler abstraction anymore?

Reply:
I do think we want a scheduler yes, but I don't think what we had could be called scheduler and at that point it just introduced extra indirection which is why I removed it. We can keep it if you think otherwise, but unless you do I suggest we introduce scheduler when it actually does scheduling

Reply:
we could easily introduce a queue on the infra side to allow retries out of the box, so that was the main advantage I was seeing with current implementation. But no strong feelings